crypto/mem_sec.c

```diff
@@ -433,8 +437,13 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
```

Comment:
One can wonder if it would be appropriate to add `defined(__linux)` to this condition...